otel/propagator_test.go

```diff
@@ -28,37 +28,31 @@ type transactionTestContext struct {
 	sampled sentry.Sampled
 }
 
-type otelTestContext struct {
-	traceId    string
-	spanId     string
-	traceFlags trace.TraceFlags
-}
-
-func createTransactionAndMaybeSpan(transactionContext transactionTestContext, withSpan bool) {
-	client, _ := sentry.NewClient(sentry.ClientOptions{
-		Dsn:           "https://abc@example.com/123",
-		Environment:   "testing",
-		Release:       "1.2.3",
-		EnableTracing: true,
-	})
-	hub := sentry.NewHub(client, sentry.NewScope())
-	ctx := sentry.SetHubOnContext(context.Background(), hub)
-
+func createTransactionAndMaybeSpan(transactionContext transactionTestContext, withSpan bool) trace.SpanContextConfig {
 	transaction := sentry.StartTransaction(
-		ctx,
+		emptyContextWithSentry(),
 		transactionContext.name,
 		sentry.SpanSampled(transactionContext.sampled),
 	)
-
 	transaction.TraceID = TraceIDFromHex(transactionContext.traceID)
 	transaction.SpanID = SpanIDFromHex(transactionContext.spanID)
-	transaction.SetDynamicSamplingContext(sentry.DynamicSamplingContextFromTransaction(transaction))
 
-	sentrySpanMap.Set(trace.SpanID(transaction.SpanID), transaction)
 	if withSpan {
 		span := transaction.StartChild("op")
+		// We want the child to have the SpanID from transactionContext, so
+		// we "swap" span IDs from the transaction and the child span.
+		transaction.SpanID = span.SpanID
```

Comment:
Not sure I fully understand this.

Reply:
When `withSpan` is passed, we want to create a transation and a child span. We also want the child span to have the predefined SpanID (passed via `transactionContext`). It means that we have to update the SpanID for the transaction. We can generate a new random one, or we can just take the already-generated one from the child span (`span`), that will be replaced with `SpanIDFromHex(transactionContext.spanID)` in the next line anyway.